extensions/wikia/AdEngine/js/MessageListener.js

```diff
@@ -55,7 +55,9 @@ define('ext.wikia.adEngine.messageListener', [
 					log(['Event matching', msg, callback], 'debug', logGroup);
 
 					callback.fn(JSON.parse(msg.data).AdEngine);
-					callbacks.splice(i, 1);
+					if (!callback.match.infinite) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)